service/src/main/java/com/epam/rft/atsy/service/impl/ApplicationsServiceImpl.java

```diff
@@ -39,6 +39,7 @@
   private ChannelRepository channelRepository;
 
 
+  @Transactional
   @Override
   public Long saveOrUpdate(ApplicationDTO applicationDTO) {
```

Comment:
Because of the new transactional annotation the actual db write happens after the method execution, so the DataIntegrityViolationException is thrown after the method and it won't be converted into DuplicateRecordException.

Quick fix is to change channelRepository.save to channelRepository.saveAndFlush

Be aware that saveAndFlush should be used as a last repository statement of a service method, otherwise it would have huge impact on concurrent access to database.